app/Http/Controllers/ExportController.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Rogue\Http\Controllers;
+
+use Rogue\Models\Signup;
+use Rogue\Services\Registrar;
+
+class ExportController extends Controller
+{
+    /**
+     * Registrar instance
+     *
+     * @var Rogue\Services\Registrar
+     */
+    protected $registrar;
+
+    /**
+     * Instantiate a new ExportController instance.
+     *
+     * @param Rogue\Services\Registrar $registrar
+     */
+    public function __construct(Registrar $registrar)
+    {
+        $this->registrar = $registrar;
+    }
+
+    /**
+     * Download the export of signup details for the specified campaign.
+     *
+     * @param int $campaignId
+     */
+    public function show($campaignId)
```

Comment:
I would love for us to extract the logic for this particular exported file into an `ExportService` If we wanted to export a file differently for something else we couldn't use this `show@ExportController` method, so if you could call something here like `$this->exportService->exportSignups(data)` we could abstract some of the business logic tied to the truth campaign needs from this route.

Reply:
@sbsmith86 I'm into that idea! Would I be extracting the "Compile the data" part out of this method to get the data and then still build the file in here? That was we can make the data be whatever we want each time but keep the file building more centralized?

Reply:
Yea I think the actual work of what is in the file, columns and such should be extracted. So all you have to do is pass an array of signups, and get the file.

Reply:
Sorry i feel like the CSV logic should also be pulled out. I don't think the controller should care about how to build a CSV, especially if we go with @DFurnes suggestion of using a third-party package for it.

Reply:
ooooo I think I misunderstood your other response! so the `return response($output, 200, $headers);` part stays the same but the file gets built in the `ExportService`?

Reply:
yea. Just having a service that has all the logic around building an export file, would be nice. You could have a method in the service class that just takes in an array of data and spits out a file. I might be nitpicking, but just trying to keep things a little more de-coupled.